### PR TITLE
Javadoc compliance with Java 8 for 2.3

### PR DESCRIPTION
**Pull request by nawroth**

### CODE REVIEW

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/GraphAlgoFactory.java [hunk 1]
```diff
@@ -375,11 +375,11 @@
     /**
      * See {@link #dijkstra(PathExpander, CostEvaluator)} for documentation
      *
-     * Instead of finding all shortest paths with equal cost, find the top {@param numberOfWantedPaths} paths.
+     * Instead of finding all shortest paths with equal cost, find the top {@code numberOfWantedPaths} paths.
      * This is usually slower than finding all shortest paths with equal cost.
      *
      * Uses a cost evaluator which uses the supplied property key to
-     * represent the cost (values of type <bold>double</bold>).
+     * represent the cost (values of type <b>double</b>).
```
Comment (thobe) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/GraphAlgoFactory.java:382]: while this change is correct from an html point of view, I wonder if we might not actually want to format this as code...

**Comment by thobe** [comment]
Looks good. Was this really all the places where the javadocs were not Java 8 compatible?

**Comment by nawroth** [comment]
@thobe No, we already got the changes I did to 2.2 in: https://github.com/neo4j/neo4j/pull/5308
Also, I set the default doclint to only care about references being resolvable.
It's only what ends up in the `neo4j-javadocs` modules that goes all-in with the doclint checks!

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/GraphAlgoFactory.java (hunk 1 above)
Comment (nawroth) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/GraphAlgoFactory.java:382]: true

**Pull request merged**